optimize.rb

```diff
@@ -79,7 +79,7 @@ def line(name, old_size, new_size)
   # Escape %, \n, and \r.
   # Ref: https://github.community/t/set-output-truncates-multiline-strings/16852/3
   summary.gsub!(/[%\n\r]/, "%" => "%25", "\n" => "%0A", "\r" => "%0D")
-  puts "::set-output name=summary::#{summary}"
+  puts "echo ':name=summary::#{summary}' >> $GITHUB_OUTPUT"
```

Comment:
@AfzalivE I don't think this will produce the intended result.

Your change makes print a command and we want the `puts` output to go to a file that's located on a path stored in `GITHUB_OUTPUT` environment variable.

I can try to propose a change later on.

Reply:
@AfzalivE this may work:

```suggestion
  File.open(ENV["GITHUB_OUTPUT"], 'a') do |fo|
    fo.puts "summary=#{summary}"
  end
```

Reply:
You are correct of course. Lol, that's what I get for copying from GitHub 😅.

Took another attempt at it, it's a little different from your suggestion above. Why do we need to redirect stdout, is it only because we're using `puts` there?

Reply:
@AfzalivE sorry, i started with the dumber thing i could find on SO and evolved from there. Your solution and mine converged 😅  
thanks for working on this!